axelrod/tests/unit/test_heterogeneous.py

```diff
@@ -0,0 +1,364 @@
+        mp = MoranProcess(players, match_class=MassBaseMatch, mode="bd", interaction_graph=graph, seed=3)
+        self.assertEqual(mp.death(0), 0)
+
+    def test_birth_in_db(self):
+        players = axl.Cooperator(), axl.Defector(), axl.TitForTat()
+        masses = [10, 20, 10]
+        set_player_mass(players, masses)
+        mp = MoranProcess(players, match_class=MassBaseMatch, mode="db", seed=1)
+        self.assertEqual(mp.death(), 2)
+        self.assertEqual(mp.birth(0), 2)
+
+    def test_birth_in_bd(self):
+        players = axl.Cooperator(), axl.Defector(), axl.TitForTat()
+        masses = [1, 1, 1]
+        set_player_mass(players, masses)
+        mp = MoranProcess(players, match_class=MassBaseMatch, mode="bd", seed=2)
+        self.assertEqual(mp.birth(), 0)
+
+    def test_fixation_check(self):
+        players = axl.Cooperator(), axl.Cooperator()
+        masses = [10, 20]
+        set_player_mass(players, masses)
+        mp = MassBasedMoranProcess(players, match_class=MassBaseMatch)
+        self.assertTrue(mp.fixation_check())
+        players = axl.Cooperator(), axl.Defector()
+        mp = MassBasedMoranProcess(players, match_class=MassBaseMatch)
+        self.assertFalse(mp.fixation_check())
+
+    def test_next(self):
+        players = axl.Cooperator(), axl.Defector()
+        masses = [10, 20]
+        set_player_mass(players, masses)
+        mp = MassBasedMoranProcess(players, match_class=MassBaseMatch)
+        self.assertIsInstance(next(mp), axl.MoranProcess)
+
+    def test_matchup_indices(self):
+        players = axl.Cooperator(), axl.Defector()
+        masses = [10, 20]
+        set_player_mass(players, masses)
+        mp = MassBasedMoranProcess(players, match_class=MassBaseMatch)
+        self.assertEqual(mp._matchup_indices(), {(0, 1)})
+
+        players = axl.Cooperator(), axl.Defector(), axl.TitForTat()
+        masses = [10, 20, 10]
+        set_player_mass(players, masses)
+        edges = [(0, 1), (2, 0), (1, 2)]
+        graph = axl.graph.Graph(edges, directed=True)
+        mp = MassBasedMoranProcess(players, match_class=MassBaseMatch, mode="bd", interaction_graph=graph)
+        self.assertEqual(mp._matchup_indices(), {(0, 1), (1, 2), (2, 0)})
+
+    def test_fps(self):
+        players = axl.Cooperator(), axl.Defector()
+        masses = [10, 20]
+        set_player_mass(players, masses)
+        mp = MoranProcess(players, match_class=MassBaseMatch, seed=1)
+        self.assertEqual(mp.fitness_proportionate_selection([0, 0, 1]), 2)
+        self.assertEqual(mp.fitness_proportionate_selection([1, 1, 1]), 2)
+        self.assertEqual(mp.fitness_proportionate_selection([1, 1, 1]), 0)
+
+    def test_exit_condition(self):
+        p1, p2 = axl.Cooperator(), axl.Cooperator()
+        masses = [10, 20]
+        set_player_mass([p1, p2], masses)
+        mp = MassBasedMoranProcess((p1, p2), match_class=MassBaseMatch)
+        mp.play()
+        self.assertEqual(len(mp), 1)
```

Comment:
> This does not need to be the MassBasedMatch right? It can just be something simple

Correct, I just used the ` MassBasedMatch` as I was already using that one.